test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ConnectionAdapterTests.cs

```diff
@@ -77,6 +78,70 @@ public class ConnectionAdapterTests
         }
 
         [Fact]
+        public async Task ImmediateFinAfterOnConnectionAsync()
```

Comment:
That it does't throw or crash. Unfortunately that's hard to assert but I can change the test name.